examples/remote.c

```diff
@@ -0,0 +1,275 @@
+/*
+ * libgit2 "remote" example - shows how to modify remotes for a repo
+ *
+ * Written by the libgit2 contributors
+ *
+ * To the extent possible under law, the author(s) have dedicated all copyright
+ * and related and neighboring rights to this software to the public domain
+ * worldwide. This software is distributed without any warranty.
+ *
+ * You should have received a copy of the CC0 Public Domain Dedication along
+ * with this software. If not, see
+ * <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+
+#include "common.h"
+
+/**
+ * This is a sample program that is similar to "git remote".  See the
+ * documentation for that (try "git help remote") to understand what this
+ * program is emulating.
+ *
+ * This demonstrates using the libgit2 APIs to modify remotes of a repository.
+ */
+
+enum subcmd {
+	subcmd_add,
+	subcmd_remove,
+	subcmd_rename,
+	subcmd_seturl,
+	subcmd_show,
+};
+
+struct opts {
+	enum subcmd cmd;
+
+	/* for command-specific args */
+	int argc;
+	char **argv;
+};
+
+static int cmd_add(git_repository *repo, struct opts *o);
+static int cmd_remove(git_repository *repo, struct opts *o);
+static int cmd_rename(git_repository *repo, struct opts *o);
+static int cmd_seturl(git_repository *repo, struct opts *o);
+static int cmd_show(git_repository *repo, struct opts *o);
+
+static void parse_subcmd(
+	struct opts *opt, int argc, char **argv);
+static void usage(const char *msg, const char *arg);
+
+int main(int argc, char *argv[])
+{
+	int retval = 0;
+	struct opts opt = {0};
+	git_repository *repo = NULL;
+
+	parse_subcmd(&opt, argc, argv);
+
+	git_libgit2_init();
+
+	check_lg2(git_repository_open_ext(&repo, ".", 0, NULL),
+		"Could not open repository", NULL);
```

Comment:
Why `_open_ext()`? If you want to simulate the git-remote command, it should call `git_repository_discover()` first so it works in subdirectories.